content/docs/user-guide/experiment-management/hydra-composition.md

```diff
@@ -0,0 +1,260 @@
+# Hydra Composition
+
+_New in DVC 2.25.0 (see `dvc version`)_
+
+In this guide we explain how to use [Hydra](https://hydra.cc/) composition for
+configuring DVC <abbr>experiments</abbr>.
+
+## How it works
+
+On each `dvc exp run` call, DVC will use Hydra to **compose** a single
+configuration object and **dump** it to `params.yaml`. This will happen
+**before** the experiment starts running.
+
+This allows to combine Hydra composition and DVC <abbr>parameters</abbr> in
+order to configure DVC <abbr>pipelines</abbr>. DVC pipelines can run multiple
+steps of different shell commands, instead of a single Python script, and use
+features like
+[Templating](/doc/user-guide/project-structure/dvcyaml-files#templating) and
+[`foreach`](/doc/user-guide/project-structure/dvcyaml-files#foreach).
+
+## Setting up Hydra
+
+<admon type="tip">
+
+Learn more about setting up Hydra in the
+[official Hydra tutorial](https://hydra.cc/docs/tutorials/basic/your_first_app/composition/).
+
+</admon>
+
+In order to start using Hydra composition capabilities it is required to have a
+directory of
+[Hydra Config Groups](https://hydra.cc/docs/tutorials/basic/your_first_app/config_groups/)
+(`conf`):
+
+```dvc
+$ tree conf
+conf
+├── config.yaml
+├── dataset
+│   ├── imagenette.yaml
+│   └── imagewoof.yaml
+└── train
+    ├── baseline.yaml
+    ├── model
+    │   ├── alexnet.yaml
+    │   ├── efficientnet.yaml
+    │   └── resnet.yaml
+    └── optimizer
+        ├── adam.yaml
+        └── sgd.yaml
+```
+
+Along with a file defining the top-level
+[Hydra Defaults List](https://hydra.cc/docs/tutorials/basic/your_first_app/defaults/)
+(`conf/config.yaml`):
+
+```dvc
+$ cat conf/config.yaml
+defaults:
+  - dataset: imagenette
+  - train/model: resnet
+  - train/optimizer: SGD
+```
+
+<admon type="info">
+
+You can set the [`dvc config hydra`](/doc/command-reference/config#hydra)
+options to provide custom locations for the _Config Group_ and the top-level
+_Defaults List_.
+
+</admon>
+
+The main difference with the official tutorial is that, instead of relying on
+the
+[`@hydra.main`](https://hydra.cc/docs/tutorials/basic/your_first_app/simple_cli/)
+Python decorator, DVC will take care of composing and dumping the configuration
+so we can instead rely on a single file: `params.yaml`.
```

Comment:
Not a blocker: For existing Hydra users, it would be nice to clarify that `dvc.api.params_show` can be pretty much a drop-in replacement for `@hydra.main`, as you show below in the Python API example.

Reply:
I didn't want to repeat so I added a link to the example section below